### DISCUSSION TITLE
Strava导出数据出问题

### DISCUSSION THREAD

**Discussion by XiXiHaHei** [yihong0618/running_page#296] [Q&A] [answered]
各位好：

首先感谢大家做了这么一个应用，很有趣，即使我这样一个零基础的编程者都想试着跟着做一做。
我在试着用的过程中碰到一个问题，就是在导出Strava数据，生成refresh token的时候，总是失败。以下是我做的过程：（_因为我是一个对编程基本零基础的人，基本指令的含义都不是很清楚，只是在复制粘贴，可能是很基本的问题，还请大家指教一下_）：

- 把read me中的指令复制，换成自己的信息，如下：
<img width="424" alt="image" src="https://user-images.githubusercontent.com/110827930/183924001-2ae53d2f-1500-4c70-848c-7b9d04999ed7.png">

- 把复制的指令直接粘贴在CMD窗口里，得到的结果如下：
<img width="482" alt="image" src="https://user-images.githubusercontent.com/110827930/183924451-0586bc9f-9072-4f3c-8dbf-2478044b66f7.png">

请大家帮忙看一下。

**Comment by yihong0618** [comment, marked as answer]
试试用个非 windows cmd 的 terminal

**Reply by yihong0618** [reply]
@shaonianche

**Comment by ben-29** [comment]
cmd不支持转义换行, 你执行的一条完整的命令被拆成多条命令不完整的命令，所以出错了。
最快的解决办法，把命令中的 "\" 和 换行 删掉，变成一行命令，再执行。
如：
`
curl -X POST https://www.strava.com/oauth/token -F client_id=${Your Client ID} -F client_secret=${Your Client Secret} -F code=${Your Code} -F grant_type=authorization_code
`
@XiXiHaHei

**Reply by XiXiHaHei** [reply]
感谢回复。

我按照您的说法，试过了，但是都不太成功，如下面两个截图。我再去试试不用CMD，找一个别的非 windows cmd 的 terminal。
<img width="530" alt="image" src="https://user-images.githubusercontent.com/110827930/184109957-6ca375a7-af21-4a5e-afa3-aae9e2e3eded.png">
<img width="539" alt="image" src="https://user-images.githubusercontent.com/110827930/184110048-43cf7d2e-ff2a-454c-83f9-ee1621fd8660.png">

**Reply by shaonianche** [reply]
> 感谢回复。
> 
> 我按照您的说法，试过了，但是都不太成功，如下面两个截图。我再去试试不用CMD，找一个别的非 windows cmd 的 terminal。 <img alt="image" width="530" src="https://user-images.githubusercontent.com/110827930/184109957-6ca375a7-af21-4a5e-afa3-aae9e2e3eded.png"> <img alt="image" width="539" src="https://user-images.githubusercontent.com/110827930/184110048-43cf7d2e-ff2a-454c-83f9-ee1621fd8660.png">

看起来curl执行了，但是没返回数据。client_id、client_secret、code=后面去掉${}符号，只填写数值试试。

**Reply by XiXiHaHei** [reply]
您看第二张图片就是去掉${}符号，只填写数值的结果，也是不行。我也试了一下直接在浏览器里访问 https://www.strava.com/oauth/token， 结果也是访问不了的。是不是因为我访问不了这个网页的原因？
<img width="686" alt="image" src="https://user-images.githubusercontent.com/110827930/184117295-c3c430a3-31bb-4718-91b4-f527cbab8fe5.png">

**Reply by yihong0618** [reply]
你这是没登陆吧

**Reply by XiXiHaHei** [reply]
登录之后也是一样的结果：
<img width="888" alt="image" src="https://user-images.githubusercontent.com/110827930/184120929-6c76580d-9643-4b81-95cb-79f0ada7ba08.png">

另外，我暂时没找到别的terminal，所以试在还是在CMD之后，把指令改成如下：
curl -X POST https://www.strava.com/oauth/token -F ‘client_id=myid’ -F ‘client_secret=mysecret’ -F ‘code=mycode’ -F ’grant_type=authorization_code‘

执行后是下面的结果：
<img width="670" alt="image" src="https://user-images.githubusercontent.com/110827930/184121334-9c2c0c28-e038-487f-b9f5-23ee1e869b57.png">

**Reply by yihong0618** [reply]
你有走这个步骤么 `Open after successful Signin [Strava Developers](http://developers.strava.com/) -> [Create & Manage Your App](https://strava.com/settings/api)`

或者说你 3 个 token 神奇成功了么？

**Reply by XiXiHaHei** [reply]
对，前面都是按照您的那个readme文件一步一步走下来的，只是到refresh token这一步走不下去了。
<img width="500" alt="image" src="https://user-images.githubusercontent.com/110827930/184128198-06e9c442-11a0-4b2e-ab31-6dc612d46a2d.png">

**Reply by XiXiHaHei** [reply]
感谢各位，应该还是因为我不会用CMD造成的问题，我找了另外一个叫postman的平台，把信息发过去了，就可以得到refresh token了。
多谢大家的支持。
另外，如果大家如果编程基础不好的，推荐用postman这个来获取refresh token.

**Reply by yihong0618** [reply]
哈哈哈，其实程序员都用 postman.
